### PR TITLE
Refactor lifecycle module names

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Refactors naming in the lifecycle file and related content script messenger API
- Changes:
  - Extension Point -> Starter Brick
  - Extension -> Mod Component  
  - Editor -> Draft
  - Persisted -> Activated
  - "tab" -> "frame" to clarify lifecycle runs in the context of a single frame
  - Use "install"/"run" terminology to match terminology in the Starter Brick interface: `StarterBrick.install`, `StarterBrick.runModComponents`
  - Use "remove" terminology to match terminology in the Starter Brick interface: `StarterBrick.removeModComponent`

## Future

- For the PR, I erred on the side of naming consistency. There's some terminology we'll likely want to clean up in the future in the Starter Brick and lifecycle APIs
  - Move away from "install" terminology, because it can be confused with mod activation. Potential candidates: "mount" (confused with React lifecycle?), "attach", "initialize"
  - Move away from "run" terminology, because for some mod components, e.g., buttons/triggers, "running" the component means attaching it to the page, but the pipeline is not run until a user action/event. Potential candidates: "load", "initialize"
  - Move away from "remove" terminology, because it can be confused with mod deactivation
  - Move away from "reload" terminology, because it can be confused with window.load event terminology
- Rename the "Reactivate" brick external-facing terminology 

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/contentScript/lifecycle.ts
Comment (grahamlangford) [inline src/contentScript/lifecycle.ts:1]: See https://github.com/pixiebrix/pixiebrix-extension/pull/8708/files#r1657130898 regarding persisted vs activated

File: src/contentScript/lifecycle.ts [hunk 1]
```diff
@@ -54,42 +54,48 @@ import pDefer from "p-defer";
...
  *
- * Mutually exclusive with _persistedExtensions.
+ * Mutually exclusive with _persistedModComponentStarterBrickMap.
  *
- * @see _persistedExtensions
+ * @see _persistedModComponentStarterBrickMap
  */
 // eslint-disable-next-line local-rules/persistBackgroundData -- Unused there
-const _editorExtensions = new Map<UUID, StarterBrick>();
+const _draftModComponentStarterBrickMap = new Map<UUID, StarterBrick>();
```
Comment (grahamlangford) [inline src/contentScript/lifecycle.ts:86]: I like this name change. 👍

File: src/contentScript/messenger/api.ts [hunk 2]
```diff
@@ -89,22 +89,22 @@ export const runBrick = getMethod("RUN_BRICK");
...
+export const runStarterBrickReaderPreview = getMethod(
+  "RUN_STARTER_BRICK_READER_PREVIEW",
 );
-export const removeInstalledExtension = getNotifier(
-  "REMOVE_INSTALLED_EXTENSION",
+export const ensureStarterBricksInstalled = getMethod(
+  "ENSURE_STARTER_BRICKS_INSTALLED",
+);
+export const getRunningStarterBricks = getMethod("GET_RUNNING_STARTER_BRICKS");
+export const removePersistedModComponent = getNotifier(
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/contentScript/messenger/api.ts:99]: Potentially — I was trying to draw the distinction between draft vs. persisted mods. "Activated" sounds more ambiguous to me.

File: src/background/removeExtensionForEveryTab.ts
Comment (grahamlangford) [inline src/background/removeExtensionForEveryTab.ts:1]: Consider renaming file

File: src/contentScript/messenger/api.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (twschiller) [reply src/contentScript/messenger/api.ts:102]: I'll go ahead and make the change. We can also switch/clarify later

File: src/contentScript/messenger/api.ts [hunk 3]
```diff
@@ -19,18 +19,21 @@
 import { getMethod, getNotifier } from "webext-messenger";
 import { forEachTab } from "@/utils/extensionUtils";
 
-export const queueReactivateTab = getNotifier("QUEUE_REACTIVATE_TAB");
+export const queueReloadFrameMods = getNotifier("QUEUE_RELOAD_FRAME_MODS");
```
Comment (grahamlangford) [inline src/contentScript/messenger/api.ts:22]: Much clearer name. 👍

File: src/contentScript/messenger/api.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (grahamlangford) [reply src/contentScript/messenger/api.ts]: That's true, but I'd prefer consistency above all. We're using draft and activated in buildAndValidateMod for instance. If we think persisted is actually better than activated, we should make that change everywhere. I think engineers will be confused and think `persisted` is a third concept

File: src/contentScript/lifecycle.ts [hunk 4]
```diff
@@ -104,50 +110,51 @@ const navigationListeners = new ReusableAbortController();
...
-    // `extensionPoint.install` should short-circuit return false if it's not available. But be defensive.
+  if (!(await starterBrick.isAvailable())) {
+    // `starterBrick.install` should short-circuit return false if it's not available. But be defensive.
     return;
   }
 
   let installed = false;
 
-  // Details to make it easier to debug extension point lifecycle
+  // Details to make it easier to debug starer brick lifecycle
```
Comment (grahamlangford) [inline src/contentScript/lifecycle.ts:138]: NIT: 
```suggestion
  // Details to make it easier to debug starter brick lifecycle
```

File: src/contentScript/pageEditor/draft/removeDraftModComponents.ts [hunk 5]
```diff
@@ -15,21 +15,21 @@
...
 
 /**
- * A version of `clearEditorExtension` that takes an object instead of a positional UUID argument.
- * @param uuid the uuid of the mod component, or null to clear all page editor mod components
- * @see clearEditorExtension
+ * A version of `removeDraftModComponent` that takes an object instead of a positional UUID argument.
+ * @param uuid an optional mod component id, or null/undefined to clear all page editor mod components
  */
-export async function clearDraftModComponents({
+// XXX: why does this method need to exist? Because messenger expects an object?
```
Comment (grahamlangford) [inline src/contentScript/pageEditor/draft/removeDraftModComponents.ts:26]: Does the messenger expect an object?

**Review by grahamlangford — APPROVED** [review]
Looks great. Much easier to reason about with some of these name changes.

For the install/uninstall debate, I'm partial to attach/detach. I've run the concept through ChatGPT for other ideas, but attach still seems to be the best I've seen that's not overloading terms we use elsewhere. The few places we use `attach` in the existing codebase seem fairly consistent with the behavior we mean when attaching a starter brick

File: src/contentScript/pageEditor/draft/removeDraftModComponents.ts (hunk 5 above)
Comment (twschiller) [reply src/contentScript/pageEditor/draft/removeDraftModComponents.ts]: Looks like it can take a primitive. I've dropped this file

**Pull request merged**